weave

```diff
@@ -975,21 +976,20 @@ dns_args() {
         fi
     fi
     if [ -z "$DNS_SEARCH_SPECIFIED" ] ; then
-      if [ -z "$HOSTNAME_SPECIFIED" ] ; then
-        DNS_ARGS="$DNS_ARGS --dns-search=$(dns_domain)"
-      else
-        DNS_ARGS="$DNS_ARGS --dns-search=."
-      fi
+      DNS_DOMAIN=$([ -z "$HOSTNAME_SPECIFIED" ] && dns_domain ||  echo ".")
```

Comment:
This doesn't seem right; when a user specifies a hostname, it should be that hostname, including any specified domains, which appears in /etc/hosts.